bentoml/_internal/frameworks/catboost.py

```diff
@@ -0,0 +1,273 @@
+            The name to give to the model in the BentoML store. This must be a valid
+            :obj:`~bentoml.Tag` name.
+        model (:obj:`~catboost.CatBoost`):
+            The CatBoost model to be saved.
+        signatures (``dict[str, ModelSignatureDict]``, optional):
+            Signatures of predict methods to be used. If not provided, the signatures default to
+            ``{"predict": {"batchable": False}}``. See :obj:`~bentoml.types.ModelSignature` for more
+            details.
+        labels (``dict[str, str]``, optional):
+            A default set of management labels to be associated with the model. An example is
+            ``{"training-set": "data-1"}``.
+        custom_objects (``dict[str, Any]``, optional):
+            Custom objects to be saved with the model. An example is
+            ``{"my-normalizer": normalizer}``.
+
+            Custom objects are currently serialized with cloudpickle, but this implementation is
+            subject to change.
+        metadata (``dict[str, Any]``, optional):
+            Metadata to be associated with the model. An example is ``{"max_depth": 2}``.
+
+            Metadata is intended for display in model management UI and therefore must be a default
+            Python type, such as ``str`` or ``int``.
+    Returns:
+        :obj:`~bentoml.Tag`: A :obj:`tag` with a format `name:version` where `name` is the
+        user-defined model's name, and a generated `version` by BentoML.
+
+    Example:
+
+    .. code-block:: python
+
+        import bentoml
+        import numpy as np
+
+        from catboost import CatBoostClassifier, Pool
+
+        # initialize data
+        train_data = np.random.randint(0, 100, size=(100, 10))
+
+        train_labels = np.random.randint(0, 2, size=(100))
+
+        test_data = catboost_pool = Pool(train_data, train_labels)
+
+        model = CatBoostClassifier(iterations=2,
+                                   depth=2,
+                                   learning_rate=1,
+                                   loss_function='Logloss',
+                                   verbose=True)
+        # train the model
+        model.fit(train_data, train_labels)
+
+        # save the model to the BentoML model store
+        bento_model = bentoml.catboost.save_model("my_catboost_model", model)
+    """  # noqa: LN001
+    if not isinstance(model, cb.CatBoost):
+        raise TypeError(f"Given model ({model}) is not a catboost.CatBoost.")
+
+    context: ModelContext = ModelContext(
+        framework_name="catboost",
+        framework_versions={"catboost": get_pkg_version("catboost")},
+    )
+
+    if signatures is None:
+        logger.info(
+            'Using default model signature `{"predict": {"batchable": False}}` for CatBoost model'
+        )
+        signatures = {
+            "predict": {"batchable": False},
+        }
+
+    if metadata is None:
+        metadata = {}
+
+    metadata[MODEL_TRAINING_CLASS_NAME_KEY] = model.__class__.__name__
```

Comment:
Metadata is something purely user defined. We should be using `model.options` here instead of `model.metadata`. Transformers, Keras, and Tensorflow have good examples of using options.